logback-classic/src/test/java/ch/qos/logback/classic/pattern/ThrowableProxyConverterTest.java

```diff
@@ -206,18 +206,95 @@ public void shouldLimitTotalLinesExcludingSkipped() throws Exception {
         String[] lines = result.split(CoreConstants.LINE_SEPARATOR);
         assertThat(lines).hasSize(3 + 1);
     }
+    
+    @Test
+    public void nullMessage() {
+        /*
+        Ensures that a null message is not sent to the output.
+        */
+        Exception t = new Exception();
+        verify(t);
+    }
+
+    @Test
+    public void circular() {
+        Exception w = null;
+        try {
+            someMethod();
+        } catch (Exception e) {
+            w = new Exception("Wrapper", e);
+            e.initCause(w);
+        }
+        verify(w);
+    }
+
+    @Test
+    public void suppressedCircular() throws Exception {
```

Comment:
@NMcCloud that's fine. It looks like @tony19 is experimenting with a somewhat chatty quality control tool.